dashboard/utils.py

```diff
@@ -277,6 +282,47 @@ def count_courses_passed(self):
                 passed_courses.add(self.edx_key_course_map[course_id])
         return len(passed_courses)
 
+    def get_pearson_exam_status(self):  # pylint: disable=too-many-return-statements
+        """
+        Get the pearson exam status for the user / program combo
+        """
+        if self.program.exam_series_code is None:
```

Comment:
This is one of those cases where this could have an empty value so I think you need to change this is `if not self.program.exam_series_code` so it bails on both `None` and `""`. Also, this needs to bail if `course.exam_module` is empty. Amir wrote some code that encapsulates these conditions [here](https://github.com/mitodl/micromasters/blob/master/exams/utils.py#L24-L31) that could probably be extracted into it's own function. **note: I don't think you want the paid condition in this common function, since Rob's screenshots including prompting the user to pay** If they haven't paid, they won't have an `ExamAuthorization` record, which I believe you covered in logic below.